### PR TITLE
feat: feedback endpoint

### LINKED ISSUE #33: Landing Page components

### PR DESCRIPTION
**Pull request by MohammedMahani**

### CODE REVIEW

File: server/routes/feedBackRouter.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const feedBackRouter = require('express').Router();
+const {
+  getAllFeedBack, addFeedBack, updateFeedBack, deleteFeedBack,
+} = require('../controllers/feedBack');
+const checkAuth = require('../middlewares/checkauth');
+
+feedBackRouter.post('/addFeedback/:commentingId', checkAuth, addFeedBack);
```
Comment (devstm) [inline server/routes/feedBackRouter.js]: When u call this router you call it with ```feedback``` prefix,
We will suffice with this prefix, and we do not need other additional words, such as ```addFeedback, allFeedback, deleteFeedback```

File: server/controllers/feedBack/getAllFeedBack.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const FeedBack = require('../../models/feedBack');
+
+const getAllFeedBack = async (req, res, next) => {
+  try {
+    const { commentingId } = req.params;
+    const data = await FeedBack.findAll({
```
Comment (devstm) [inline server/controllers/feedBack/getAllFeedBack.js:6]: apply pagination

**Review by devstm — CHANGES_REQUESTED** [review]

File: server/routes/feedBackRouter.js (hunk 1 above)
Comment (abdullah-alsharif) [reply server/routes/feedBackRouter.js]: +1

File: server/models/feedBack.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const User = require('./user');
+
+const FeedBack = sequelize.define('feedbacks', {
+  comment: {
+    type: DataTypes.STRING,
+    allowNull: true,
+  },
+  star: {
+    type: DataTypes.INTEGER,
+    allowNull: false,
```
Comment (abdullah-alsharif) [inline server/models/feedBack.js:12]: the client is free to add star or not

File: server/controllers/feedBack/updateFeedBack.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const FeedBack = require('../../models/feedBack');
+const { addFeedBackSchema } = require('../../utils/validation');
+
+const updataFeedBack = async (req, res, next) => {
+  try {
+    const { commentId } = req.params;
+    const { comment, star } = req.body;
+
```
Comment (abdullah-alsharif) [inline server/controllers/feedBack/updateFeedBack.js:9]: here i want to make sure if the commmenter user is exit in the User Table

File: server/controllers/feedBack/updateFeedBack.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const FeedBack = require('../../models/feedBack');
```
Comment (abdullah-alsharif) [inline server/controllers/feedBack/updateFeedBack.js:1]: use feedback as object (you have an index file on models folder)

File: server/controllers/feedBack/updateFeedBack.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const updataFeedBack = async (req, res, next) => {
+  try {
+    const { commentId } = req.params;
+    const { comment, star } = req.body;
+
+    const validUpdate = await addFeedBackSchema.validateAsync({
+      comment, star,
+    }, { abortEarly: false });
+
+    const updatedData = FeedBack.update({ commentId, ...validUpdate }, {
```
Comment (abdullah-alsharif) [inline server/controllers/feedBack/updateFeedBack.js]: - you must be sure the data is exist before update it (you can use findByPk)
- update query is returns an array with one or two elements, use ( returning: true) option to be sure the array have two items
- make the error custome with its state (you can see updateMessage.js file on 15-biuld-database branch)

File: server/controllers/feedBack/addFeedBack.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    const commenterId = req.user.id;
+    const { commentingId } = req.params;
+    const { comment, star } = req.body;
+    const validFeedBack = await addFeedBackSchema.validateAsync({
+      comment, star,
+    }, { abortEarly: false });
+
+    const data = await FeedBack.create({ commenterId, commentingId, ...validFeedBack });
+    res.status(200).json({
```
Comment (Basel-01) [inline server/controllers/feedBack/addFeedBack.js]: use response template we agreed on hackmd

File: server/controllers/feedBack/deleteFeedBack.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+const FeedBack = require('../../models/feedBack');
+
+const deleteFeedBack = async (req, res, next) => {
+  try {
+    const { feedbackId } = req.body;
+    const feedback = await FeedBack.findByPk(feedbackId);
+    await feedback.destroy();
+    res.status(200).json({
```
Comment (Basel-01) [inline server/controllers/feedBack/deleteFeedBack.js:8]: use response template we agreed on hackmd

File: server/controllers/feedBack/getAllFeedBack.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+      where: {
+        commentingId,
+      },
+    });
+    res.status(200).json({ data });
```
Comment (Basel-01) [inline server/controllers/feedBack/getAllFeedBack.js]: use response template we agreed on hackmd

File: server/controllers/feedBack/updateFeedBack.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+      where: {
+        id: commentId,
+      },
+    });
+    res.status(200).json({
```
Comment (Basel-01) [inline server/controllers/feedBack/updateFeedBack.js]: use response template we agreed on hackmd

**Review by Basel-01 — CHANGES_REQUESTED** [review]

File: server/controllers/feedBack/updateFeedBack.js [hunk 11]
```diff
@@ -1,23 +1,34 @@
 const FeedBack = require('../../models/feedBack');
+const CustomeError = require('../../utils/helper/customeError');
```
Comment (abdullah-alsharif) [inline server/controllers/feedBack/updateFeedBack.js]: const { CustomeError  }= require('../../utils');

File: server/database/config/connection.js [hunk 12]
```diff
@@ -10,13 +10,13 @@ const sequelize = new Sequelize(url, {
...
+    require: true,
+    rejectUnauthorized: NODE_ENV !== 'development',
+  },
 });
 if (NODE_ENV === 'development') {
-  // sequelize.sync();
+  sequelize.sync();
 }
 
 module.exports = sequelize;
```
Comment (abdullah-alsharif) [inline server/database/config/connection.js]: delete file

**Review by abdullah-alsharif — CHANGES_REQUESTED** [review]

**Review by Basel-01 — APPROVED** [review]

File: server/database/config/connection.js [hunk 13]
```diff
@@ -10,13 +10,13 @@ const sequelize = new Sequelize(url, {
   dialectOptions: {
     charset: 'utf8',
   },
-  // ssl: {
-  //   require: true,
-  //   rejectUnauthorized: NODE_ENV !== 'development',
-  // },
+  ssl: {
```
Comment (devstm) [inline server/database/config/connection.js]: remove that

**Review by devstm — CHANGES_REQUESTED** [review]

**Review by devstm — APPROVED** [review]

**Review by AyaQunoo — APPROVED** [review]

**Pull request merged**